src/Stan/Config.hs

```diff
@@ -189,6 +190,36 @@ configToCliCommand ConfigP{..} = "stan " <> T.intercalate " \\\n     "
 mkDefaultChecks :: [FilePath] -> HashMap FilePath (HashSet (Id Inspection))
 mkDefaultChecks = HashMap.fromList . map (, inspectionsIds)
 
+{- | Apply configuration to the given list of files to get the set of
+inspections for each file.
+
+The algorithm:
+
+1. Remove all files specified by the @remove@ option.
+2. Run 'applyChecks' on the remaining files.
+-}
+applyConfig
+    :: [FilePath]
+    -- ^ Paths to project files
+    -> Config
+    -- ^ Stan runtime configuration
+    -> HashMap FilePath (HashSet (Id Inspection))
+    -- ^ Resulting set of inspections for each file
+applyConfig paths ConfigP{..} =
+    applyChecks (filter notRemoved paths) configChecks
+  where
+    -- TODO: can be implemented efficiently, but the more efficient
+    -- implementation is required only if the @configRemoved@ can have
+    -- >= 1K entries
```

Comment:
Don't think that this would  be the case, as know it would require >= 1K entries *by hands* :smiling_imp: